src/components/checkbox/checkbox.ts

```diff
@@ -192,6 +198,33 @@ export class MdCheckbox implements ControlValueAccessor {
     this.toggle();
   }
 
+  /**
+   * Event handler used for (focus) events.
+   */
+  onFocus(evt: Event) {
+    this._isFocused = true;
+  }
+
+  /**
+   * Event handler used for (blur) events.
+   */
+  onBlur() {
+    this._isFocused = false;
+    this.onTouched();
+  }
+
+  /**
+   * Event handler used for (keydown.space) events on the window. Used to prevent spacebar events
+   * from bubbling when the component is focused, which prevents side effects like page scrolling
+   * from happening.
+   */
+  onWindowKeyDown(evt: KeyboardEvent) {
```

Comment:
As discussed in chat: just use `keydown` on the host element since it should only fire when it has focus anyway?